packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -561,33 +580,10 @@ export class IndexedDbPersistence implements Persistence {
         if (currentLeaseIsValid) {
           if (this.isLocalClient(currentPrimary) && this.networkEnabled) {
             return true;
-          }
-
-          if (!this.isLocalClient(currentPrimary)) {
-            if (!currentPrimary!.allowTabSynchronization) {
-              // Fail the `canActAsPrimary` check if the current leaseholder has
-              // not opted into multi-tab synchronization. If this happens at
-              // client startup, we reject the Promise returned by
-              // `enablePersistence()` and the user can continue to use Firestore
-              // with in-memory persistence.
-              // If this fails during a lease refresh, we will instead block the
```

Comment:
It seems like removing this error here is a regression. Consider:
  * If the client successfully starts
  * goes into the background and loses access to IndexedDB
  * stays in the background for longer than the lease duration

If device comes back online the user opens another tab before our retry has succeeded in refreshing the lease, there's a very real chance we'll lose this race. Users don't even have to be mixing and matching modes for this to fail now.

Reply:
Summary of my last 5 minutes:
- Gil is wrong.
- Gil is right.
- Gil is wrong.
- Gil is right.
- Gil it right and wrong.

The idea behind the lease refresh was it would be a "best effort" only operation. If if fails or stalls, any operation to IndexedDB should verify the expected state. That is:

- "readwrite-primary" operations should verify that the tab has the lease or can become the leaseholder. This already happens.
- "readwrite", "readonly" operations should verify that they are allowed to access IndexedDB (i.e. there is no primary tab without `synchronizeTabs` turned on). This is not happening right now. Instead, this is periodically verified via the code that I removed.

So, in conclusion: We should move this verification and run it as part of every transaction. I will leave the code in `master` and work on this in a separate PR.